src/models.py

```diff
@@ -180,16 +184,18 @@ def forward(self, x):
         return out
 
 
-class ChristophCNNwithPCEN(PCENMixin, ChristophCNN):
+class GoogleResNet50withPCEN(GoogleResNet50):
     '''
-    same as ChristophCNN with first PCEN layer
+    same as GoogleResNet50 with first (non-trainable) PCEN layer
     '''
 
-    def __init__(self, num_classes=2, eps=1E-6, s=0.025, alpha=0.98, delta=2, r=0.5, trainable=True):
-        super().__init__(num_classes=num_classes, eps=eps, s=s, alpha=alpha, delta=delta, r=r, trainable=trainable)
+    def __init__(self, num_classes=2, eps=1E-6, s=0.025, alpha=0.98, delta=2, r=0.5, trainable=False):
+        super().__init__(num_classes=num_classes)
+        self.pcen_model = PCENTransform(eps, s, alpha, delta, r, trainable)
 
     def forward(self, x):
-        out = super().forward(x)
+        out = self.pcen_model.forward(x)
```

Comment:
I don't think you explicitly need to call forward here, since the __call__ method of nn.Module wraps the forward
simply `self.pcen_model(x)' will do